tests/utils/test_nl.c

```diff
@@ -0,0 +1,66 @@
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include "utils/nl.h"
+
+struct nl_test_state_t {
+  struct nlctx *context; // default nlctx test context
+};
+
+const char *test_nl_interface_name = "test_nl_interface_name";
+// reserved testing IP https://datatracker.ietf.org/doc/html/rfc5737
+const char *test_nl_interface_ip = "192.0.2.1";
+const char *test_nl_interface_broadcast_ip = "192.0.2.255";
+const char *test_nl_subnet_mask = "24";
+
+static int setup(void **state) {
+  struct nl_test_state_t *ptr = malloc(sizeof(struct nl_test_state_t));
+  if (ptr == NULL) {
+    return -1;
+  }
+  ptr->context = nl_init_context();
+  if (ptr->context == NULL) {
+    return -1;
+  }
+  *state = ptr;
+  return 0;
+}
+
+static int teardown(void **state) {
```

Comment:
Did you test if the interface is actually removed?

Reply:
No need :)

I've only added a basic test that checks whether `nl_set_interface_ip()` fails if the interface doesn't exist, so there's actually no interface.

I couldn't figure out how to add a test that checks whether it actually works (I'm guessing we'd need to do a bunch of mocks).